tao_compiler/mlir/disc/transforms/disc_shape_optimization_utils.h

```diff
@@ -87,6 +87,102 @@ inline bool operator!=(const SymbolicDimProduct& lhs,
 llvm::raw_ostream& operator<<(llvm::raw_ostream& os,
                               const SymbolicDimProduct& product);
 
+struct SliceOpShapeHelper {
+  // A constructor
+  // `op` should be mhlo.real_dynamic_slice or lmhlo.real_dynamic_slice
+  explicit SliceOpShapeHelper(Operation* op);
+
+  // Returns  the attribute name used when saving.
+  // An exam ple is
+  //   "mhlo .real_dynamic_slice"(...) { kDiscSliceOpStaticKnownInfo = {...}}
+  static StringRef getAttrName() { return "kDiscSliceOpStaticKnownInfo"; }
+
+  // Saves the updates attribute to the `op` this helper holds.
+  LogicalResult save();
+
+  enum {
```

Comment:
nit: a named enum class? `kUnknown` looks pretty easy to get confilct..

Reply:
Done